python/cudf/cudf/core/window/rolling.py

```diff
@@ -205,6 +208,33 @@ def _apply_agg_series(self, sr, agg_name):
                 self.center,
                 agg_name,
             )
+        elif isinstance(self.window, BaseIndexer):
+            start, end = self.window.get_window_bounds(
+                num_values=len(self.obj),
+                min_periods=self.min_periods,
+                center=self.center,
+                closed=None,
+            )
+            start = as_column(start, dtype="int32")
+            end = as_column(end, dtype="int32")
+
+            idx = cudf.core.column.arange(len(start))
+            preceding_window = (idx - start + cudf.Scalar(1, "int32")).astype(
+                "int32"
+            )
+            following_window = (end - idx - cudf.Scalar(1, "int32")).astype(
+                "int32"
+            )
+
+            result_col = libcudf.rolling.rolling(
```

Comment:
Nit: but we can set `preceding_window`, `following_window` and `window_size` in the `if`, `elif` and `else` blocks and follow with a single call to `libcudf.rolling.rolling()` instead of repeating the same call thrice. I think that would be slightly cleaner

Reply:
Changed it accordingly 👍